src/flags.c

```diff
@@ -2,3 +2,21 @@
 bool FlagParseComments;
 bool FlagShowErrorCodes;
 bool FlagVerbose = true;
+
+void parseFlag(const char *flagName, bool *out, int argc, char **argv) {
+    for (int i = 1; i < argc; i++) {
+        if (strlen(argv[i]) == 0 || *argv[i] != '-') {
+            perror("Flag was not prefixed with '-'");
+            exit(1);
+        }
+        
+        if (strcmp(argv[i] + 1, flagName) == 0) {
+            *out = true;
+            return;
+        }
+    }
+    
+    if (*out == NULL) {
```

Comment:
Not sure what this check is here as the `NULL` is just a `0` so the comparison is equivalent to `if (*out == false) *out = false;` which changes nothing.